extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/manager/impl/StructureManagerImpl.java

```diff
@@ -28,78 +30,95 @@
  *
  * @author Michael Ochel - Initial contribution
  * @author Matthias Siegele - Initial contribution
- *
  */
 public class StructureManagerImpl implements StructureManager {
 
+    private static class ZoneGroupsNameAndIDMap {
+        public final String zoneName;
+        public final int zoneID;
+
+        public final Map<Short, String> groupIdNames;
+        public final Map<String, Short> groupNameIds;
+
+        public ZoneGroupsNameAndIDMap(final int zoneID, final String zoneName, JsonArray groups) {
+            this.zoneID = zoneID;
+            this.zoneName = zoneName;
+
+            HashMap<Short, String> groupIdNames = new HashMap<Short, String>(groups.size());
```

Comment:
As we are using Java 8 we can use the diamond operator that comes with Java 7

Reply:
Any reason to first introdocue the local variable and later assign the local variable to the member variable?
Why not simplify this one by

```Java
groupIdNames = new HashMap<>(groups.size());
```

I like your size hint usage.

Reply:
I think I wasn't sure at this time, if I can put something on a final HashMap. But it works and thus `groupIdNames = new HashMap<>(groups.size());` is better. So you are right.